src/pages/iou/request/step/IOURequestStepDistanceRate.tsx

```diff
@@ -69,7 +81,13 @@ function IOURequestStepDistanceRate({
     const initiallyFocusedOption = sections.find((item) => item.isSelected)?.keyForList;
 
     function selectDistanceRate(customUnitRateID: string) {
-        IOU.updateDistanceRequestRate(transactionID, customUnitRateID, policy?.id ?? '');
+        let taxAmount;
+        if (isDistanceRequest && transaction?.amount && policy?.customUnits && customUnitID && isOffline) {
```

Comment:
`isDistanceRequest` is redundant as this step is rendered for distance requests only.
Why `isOffline` check? We should update when user is online too.
I think we should add `isTaxTrackingEnabled()` because we want to update tax details only when it is enabled.
We should also update corresponding taxRate.
taxPercentage and taxRate to use here are - 
`var taxRate = policy?.customUnits[customUnitID].rates[customUnitRateID].attributes?.taxRateExternalID ?? '';`

Reply:
Let's add test to verify this.

When user updates distance rate, then tax rate associated with distance rate should also change

Reply:
@MonilBhavsar why do we need to have FE calculations when user if online? It should be done on BE side

Reply:
@MonilBhavsar good point - updated PR